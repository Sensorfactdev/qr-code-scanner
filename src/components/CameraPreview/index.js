import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const InnerCameraPreview = styled.video`
`;

const CameraPreview = ({ source, videoRef }) => (
  <InnerCameraPreview
    autoPlay
    innerRef={(el) => { videoRef(el); }}
    src={source}
  />
);

CameraPreview.propTypes = {
  source: PropTypes.string.isRequired,
  videoRef: PropTypes.func.isRequired,
};

export default CameraPreview;
