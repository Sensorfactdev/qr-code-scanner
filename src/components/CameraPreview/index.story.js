import React from 'react';
import { storiesOf } from '@storybook/react';
import CameraPreview from './index';

storiesOf('CameraPreview', module)
  .add('default view', () => (
    <CameraPreview
      source="hlrfjkle"
      videoRef={f => f}
    />
  ));
