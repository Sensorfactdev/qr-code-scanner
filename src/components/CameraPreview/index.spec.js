import React from 'react';
import renderer from 'react-test-renderer';
import CameraPreview from './index';

describe('CameraPreview', () => {
  it('should render', () => {
    const component = renderer.create(
      <CameraPreview
        source="blob://someblobl.com"
        videoRef={f => f}
      />
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
