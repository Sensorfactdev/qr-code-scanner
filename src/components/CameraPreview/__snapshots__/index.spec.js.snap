// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraPreview should render 1`] = `
<video
  autoPlay={true}
  className="sc-bdVaJa UoSJW"
  src="blob://someblobl.com"
/>
`;
